### ISSUE TITLE
Add a way to get durations of shadow scheduler tasks

### ISSUE THREAD

**Issue by ahal** [mozilla/mozci#102]
Currently the dashboard is measuring the # of tasks scheduled, but we should probably track the total duration of all tasks instead. Some tasks are an order of magnitude longer than others and our metrics should take this into account.

The problem is that we need the total time of the "shadow scheduler" tasks, and those don't all run on the push. So we can't just sum all the `task.duration`s. We'll need to query the average duration from ActiveData or something. Since these should be somewhat stable, we also likely don't want to re-query them *too* often (maybe cache results for a day or two)..

An argument can be made that this should live in the `ETL`, though I think this data (especially a cache of average runtimes) would be useful to other consumers as well.

**Comment by marco-c** [comment]
bugbug could use this data to penalize long running tasks (e.g. if it finds a test with "medium" confidence to fail, it might decide not to run it because of the cost).

**Comment by ahal** [comment]
I guess we'll also want the durations of manifests as well eventually. Though fine to start with tasks I think.

**Comment by ahal** [comment]
I believe a good way of implementing this is to add `duration` (both total and average) to the `Label/GroupSummary` classes. Then we can implement a static `create` method on both that accepts a date range as input and uses ActiveData to find all labels/groups within the range and returns the `Summary` object containing them. So from a user's view, it might look something like:

```
summary = LabelSummary.create(from="today", to="today-week")
print(summary.duration)  # total duration of all tasks
print(summary.average_duration)
```

We could potentially support other ways of instantiating this other than a date range (i.e, a revset) if we wanted.

**Comment by klahnakoski** [comment]
>  have a helper that grabs all task durations

Yes, mozci is slow because it asks for data one-push-at-a-time. It would be faster to pull all the information from all pushes, all labels (all groups) at once.  Not sure how to do that; maybe the "helper" can do it, and fill the caches.

**Comment by marco-c** [comment]
Blocks #1460 too.

**Comment by klahnakoski** [comment]
Some steps:

1. (mozci) Get `duration` of every task
2. (mozci) Get `duration` of manifests (`max(test.end.time) - min(test.start_time)` for all test in manifest); be sure to consider tests with no manifest
3. (mozci) Calculate `overhead`: all the time not covered by manifests in the `(task.start_time, task.end_time)` range
4. (mozci) raise warnings if the manifest times overlap
5. (bigquery view)  every manifest and median duration
6. (bigquery view) average task `overhead` by ?suite?
7. (cia-tasks scipy) Optional: Calculate simple regression between number of manifests in task and `overhead`.  Use it if there is a strong signal
8. (?decision task?) Ensure shadow schedulers are emitting specific manifests they wish to schedule as an artifact
9. (cia-tasks) Ensure the `cia-tasks/scripts/etl/schedulers` is ingesting shadow schedulers manifests
10. (data studio) use stats from 5, 6, and 7 to estimate shadow scheduler time

**Comment by worldomonation** [comment]
>     2. (mozci) Get `duration` of manifests (`max(test.end.time) - min(test.start_time)` for all test in manifest); be sure to consider tests with no manifest

I can't think of one - do you happen to have an example of this?

**Comment by marco-c** [comment]
Still something to do here.

**Comment by worldomonation** [comment]
Right, thanks for reopening. I will implement rest of the `mozci` parts soon.

**Comment by armenzg** [comment]
@worldomonation is out for a bit. Someone can pick this up.
